### PR TITLE
#246 - Fix for obsolete Modal examples

### PR DESCRIPTION
**Pull request by SayusiAndo**
Basically, it is about changing `OnClick` to `@onclick` so far.
I haven't done any PR for years, let me know if I messed up something.

### CODE REVIEW

**Comment by jbomhold3** [comment]
OnClick refers to the event in the component it's self.   This event is forward to the correct HTML item. Passing @onclick at the component level will trigger the event from the root of the component.  While this may appear to work correctly in some cases it's not in all. In the case of the modal header your clicking the div vs the close button. Meaning clicking anywhere on the header will close the modal. 
See https://github.com/chanan/BlazorStrap/blob/master/src/BlazorStrap/Components/Model/BSModalHeader.razor.cs
https://github.com/chanan/BlazorStrap/blob/master/src/BlazorStrap/Components/Model/BSModalHeader.razor

For the button, the change will work fine.  We can even decrepit OnClick inside of them and remove it in a later release after giving people time to switch

File: src/SampleCore/Pages/Modals.razor [hunk 1]
```diff
@@ -16,27 +16,27 @@
 
 <h3>Live Demo</h3>
 <div class="docs-example">
-    <BSButton Color="Color.Primary" OnClick="@(() => LiveDemo.Show())">Launch demo modal</BSButton>
+    <BSButton Color="Color.Primary" @onclick="@(() => LiveDemo.Show())">Launch demo modal</BSButton>
     <BSModal @ref="LiveDemo">
-        <BSModalHeader OnClick="@(() => LiveDemo.Hide())">Modal title</BSModalHeader>
+        <BSModalHeader @onclick="@(() => LiveDemo.Hide())">Modal title</BSModalHeader>
```
Comment (jbomhold3) [inline src/SampleCore/Pages/Modals.razor]: Should not be changed.

File: src/SampleCore/Pages/Modals.razor [hunk 2]
```diff
@@ -16,27 +16,27 @@
...
 </div>
 <PrettyCode CodeFile="_content/SampleCore/snippets/modals/modals2.html" />
 
 <h3>Vertically Centered</h3>
 <div class="docs-example">
-    <BSButton Color="Color.Primary" OnClick="@onToggle">Launch demo modal</BSButton>
+    <BSButton Color="Color.Primary" @onclick="@onToggle">Launch demo modal</BSButton>
     <BSModal @ref="Centered" IsCentered="true">
-        <BSModalHeader OnClick="@onToggle">Modal title</BSModalHeader>
+        <BSModalHeader @onclick="@onToggle">Modal title</BSModalHeader>
```
Comment (jbomhold3) [inline src/SampleCore/Pages/Modals.razor]: Should not be changed.

File: src/SampleCore/Pages/Modals.razor [hunk 3]
```diff
@@ -45,22 +45,22 @@
 <h3>Sizes</h3>
 
 <div class="docs-example">
-    <BSButton Color="Color.Primary" OnClick="onclick3">Large modal</BSButton>
+    <BSButton Color="Color.Primary" @onclick="onclick3">Large modal</BSButton>
     <BSModal @bind-IsOpen="@IsOpen3" Size="Size.Large">
-        <BSModalHeader OnClick="onclick3">Modal title</BSModalHeader>
+        <BSModalHeader @onclick="onclick3">Modal title</BSModalHeader>
```
Comment (jbomhold3) [inline src/SampleCore/Pages/Modals.razor]: Should not be changed.

File: src/SampleCore/Pages/Modals.razor [hunk 4]
```diff
@@ -45,22 +45,22 @@
...
-            <BSButton Color="Color.Primary" OnClick="onclick3">Save Changes</BSButton>
+            <BSButton Color="Color.Secondary" @onclick="onclick3">Close</BSButton>
+            <BSButton Color="Color.Primary" @onclick="onclick3">Save Changes</BSButton>
         </BSModalFooter>
     </BSModal>
-    <BSButton Color="Color.Primary" OnClick="onclick4">Small modal</BSButton>
+    <BSButton Color="Color.Primary" @onclick="onclick4">Small modal</BSButton>
     <BSModal @bind-IsOpen="@IsOpen4" Size="Size.Small">
-        <BSModalHeader OnClick="onclick4">Modal title</BSModalHeader>
+        <BSModalHeader @onclick="onclick4">Modal title</BSModalHeader>
```
Comment (jbomhold3) [inline src/SampleCore/Pages/Modals.razor]: Should not be changed.

File: src/SampleCore/wwwroot/snippets/modals/modals2.html [hunk 5]
```diff
@@ -1,10 +1,10 @@
-﻿ <BSButton Color="Color.Primary" OnClick="@(() => LiveDemo.Show())">Launch demo modal</BSButton>
+﻿ <BSButton Color="Color.Primary" @onclick="@(() => LiveDemo.Show())">Launch demo modal</BSButton>
     <BSModal @ref="LiveDemo">
-        <BSModalHeader OnClick="@(() => LiveDemo.Hide())">Modal title</BSModalHeader>
+        <BSModalHeader @onclick="@(() => LiveDemo.Hide())">Modal title</BSModalHeader>
```
Comment (jbomhold3) [inline src/SampleCore/wwwroot/snippets/modals/modals2.html]: Should not be changed.

File: src/SampleCore/wwwroot/snippets/modals/modals3.html [hunk 6]
```diff
@@ -1,10 +1,10 @@
-﻿<BSButton Color="Color.Primary" OnClick="@onToggle">Launch demo modal</BSButton>
+﻿<BSButton Color="Color.Primary" @onclick="@onToggle">Launch demo modal</BSButton>
 <BSModal @ref="Centered" IsCentered="true">
-    <BSModalHeader OnClick="@onToggle">Modal title</BSModalHeader>
+    <BSModalHeader @onclick="@onToggle">Modal title</BSModalHeader>
```
Comment (jbomhold3) [inline src/SampleCore/wwwroot/snippets/modals/modals3.html]: Should not be changed.

File: src/SampleCore/wwwroot/snippets/modals/modals4.html [hunk 7]
```diff
@@ -1,18 +1,18 @@
-﻿<BSButton Color="Color.Primary" OnClick="onclick3">Large modal</BSButton>
+﻿<BSButton Color="Color.Primary" @onclick="onclick3">Large modal</BSButton>
 <BSModal IsOpen="@IsOpen3" Size="Size.Large">
-    <BSModalHeader OnClick="onclick3">Modal title</BSModalHeader>
+    <BSModalHeader @onclick="onclick3">Modal title</BSModalHeader>
```
Comment (jbomhold3) [inline src/SampleCore/wwwroot/snippets/modals/modals4.html]: Should not be changed.

File: src/SampleCore/wwwroot/snippets/modals/modals4.html [hunk 8]
```diff
@@ -1,18 +1,18 @@
...
-        <BSButton Color="Color.Primary" OnClick="onclick3">Save Changes</BSButton>
+        <BSButton Color="Color.Secondary" @onclick="onclick3">Close</BSButton>
+        <BSButton Color="Color.Primary" @onclick="onclick3">Save Changes</BSButton>
     </BSModalFooter>
 </BSModal>
-<BSButton Color="Color.Primary" OnClick="onclick4">Small modal</BSButton>
+<BSButton Color="Color.Primary" @onclick="onclick4">Small modal</BSButton>
 <BSModal IsOpen="@IsOpen4" Size="Size.Small">
-    <BSModalHeader OnClick="onclick4">Modal title</BSModalHeader>
+    <BSModalHeader @onclick="onclick4">Modal title</BSModalHeader>
```
Comment (jbomhold3) [inline src/SampleCore/wwwroot/snippets/modals/modals4.html]: Should not be changed.

**Comment by SayusiAndo** [comment]
Thanks for the review. I'll make the changes in the morning (CET).

**Pull request merged**